historian/frontend/src/app/tag/tag-dashboard/tag-dashboard.component.ts

```diff
@@ -78,28 +79,40 @@ export class TagDashboardComponent implements OnInit {
   }
 
   onFilterQuery(query: string) {
-    this.tsListComp.dataTreeComp.search(query);
+    this.tagTreeComp.jsTree.search(query);
   }
 
   anyTagSelected(): boolean {
-    return this.selectedTags.size !== 0;
+    return this.selectedTags.length !== 0;
   }
 
   multipleTagSelected(): boolean {
-    return this.selectedTags.size > 1;
+    return this.selectedTags.length > 1;
   }
   // update tag in tree.
-  onTagSaved(tag: Tag): void {
-    const nodeToUpdate = this.tsListComp.dataTreeComp.getNode(tag.id);
+  onTagSaved(tag: IHistorianTag): void {
+    const nodeToUpdate = this.tagTreeComp.jsTree.getNode(tag.id);
     Object.assign(nodeToUpdate.original.tag, tag);
-    this.updateCreation(tag);
+    const currentType = this.tagTreeComp.jsTree.getType(nodeToUpdate);
+    if (currentType === TypesName.TAG_OPC) {
+      this.tagTreeComp.jsTree.setType(nodeToUpdate, TypesName.TAG_HISTORIAN);
+    }
+    this.selectedTags = this.selectedTags.map(tagInput => {
+      if (tagInput.tag.id === tag.id) {
+        tagInput.isCreation = false;
+        return tagInput;
+      } else {
+        return tagInput;
+      }
+    });
   }
 
-  private updateCreation(tag: ITag): void {
-    if (Tag.isHistorianTag(tag)) {
-      this.isCreation = false;
-    } else {
-      this.isCreation = true;
-    }
+  canDeactivate(): Observable<boolean> | boolean {
+    if (this.tagFormIsClean()) return true;
+    return this.dialogService.confirm('Discard changes ?');
```

Comment:
You should put message strings into a property file or at least a special class to prepare for i18n later on